sqlglot/dialects/snowflake.py

```diff
@@ -166,6 +166,15 @@ def _build_regexp_replace(args: t.List) -> exp.RegexpReplace:
     return regexp_replace
 
 
+def _build_replace(args: t.List) -> exp.Replace:
```

Comment:
Isn't the logic the same as Presto? Why do something different for Snowflake? I'd suggest extracting the Presto logic out into a helper within `dialect.py` and then importing it in both dialects, so that it can be reused elsewhere if needed.